BlueprintUI/Sources/Layout/GridRow.swift

```diff
@@ -0,0 +1,303 @@
+import UIKit
+
+/// Like `Row`, `GridRow` displays a list of items in a linear horizontal layout. Unlike `Row`, `GridRow` provides
+/// convenience for describing columnar layout.
+///
+/// Horizontally, `GridRow` children are stretched to fill the available space. Vertically, children are aligned
+/// according to the `verticalAlignment` property.
+///
+/// Children may be sized proportionally or absolutely. Proportionally-sized children are granted a proportion of
+/// the total layout space **after** absolutely-sized children and margins have been subtracted.
+///
+/// ## Example:
+///
+/// ```
+/// GridRow { row in
+///   row.verticalAlignment = .fill
+///   row.spacing = 8.0
+///   row.add(child: .proportional(0.75, name))
+///   row.add(child: .proportional(0.25, number))
+///   row.add(child: .absolute(50, status))
+/// }
+/// ```
+///
+/// ## Expected layout:
+///
+/// ```
+/// ┌────────────────────────────┬─┬────────┬─┬──────────────────┐
+/// │            name            │ │ number │ │      status      │
+/// │            (75%)           │8│  (25%) │8│     (100 pts)    │
+/// │                            │ │        │ │                  │
+/// ●──────────── 150 ───────────● ●── 50 ──● ●─────── 100 ──────●
+/// └────────────────────────────┴─┴────────┴─┴──────────────────┘
+/// ●──────────────────────────── 316 ───────────────────────────●
+/// ```
+public struct GridRow: Element {
+    // MARK: - properties -
+    /// How children are aligned vertically. By default, `.fill`.
+    public var verticalAlignment: Row.RowAlignment = .fill
+
+    /// The space between children. By default, 0.
+    public var spacing: CGFloat = 0
+
+    /// The child elements to be laid out. By default, an empty array.
+    public var children: [Child] = []
+
+    // MARK: - initialization -
+    public init(configure: (inout GridRow) -> Void = { _ in }) {
+        configure(&self)
+    }
+
+    // MARK: - mutations -
+    public mutating func add(child: Child) {
```

Comment:
Thought: I think this is probably fine (or even perhaps preferable if we were consistent about the approach), but should the values that make up `Child` be provided as separate parameters here? Maybe just as a separate convenience func?

Alternatively, should we migrate `Stack` to use this style?

No need to take action, but it might be worth driving towards a consistent API overall if there's no technical reason to stray.

Reply:
It's a great question. The main difference, I think, is that `GridRow` uses a `struct`, while `Row` uses a tuple, which makes the spelling of these sites a bit awkward. What's more, `Row` can provide reasonable defaults for `grow` and `shrink` priority, but I can't.

I will replace the existing sugar with a style more in keeping with `Row`. Does something like 

```swift
public mutating func addAbsolute(child: Element, width: CGFloat) { ... }
public mutating func addProportional(child: Element, proportion: CGFloat) { ... }
```

roughly match your expectations?

Reply:
Ah, the lack of reasonable defaults is an interesting point.

I was thinking:
```swift
public mutating func add(width: Width, key: AnyHashable? = nil, child: Element) {
    add(child: .init(width: width, key: key, element: child)
}
```
but what you have above seems fine as well.

Curious what the other Blueprint folks think, but it's probably fine to land as is and provide additional conveniences if they seem helpful 🤷

Reply:
> ```swift
> public mutating func add(width: Width, key: AnyHashable? = nil, child: Element) {
>     add(child: .init(width: width, key: key, element: child)
> }
> ```
> 
> Curious what the other Blueprint folks think, but it's probably fine to land as is and provide additional conveniences if they seem helpful 🤷

This is probably the closer to the API I would build personally, but I don't have strong feelings about it. Both of them can support new params (e.g. alignment guides) or convenience methods.

Reply:
Thanks for the suggestion. I'll update before merging.